shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -580,7 +581,165 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], newDefs: Bo
     raise(WarningReport(msg"[${cur.headOption.map(_._1).mkString}] ${""+msg}" -> loco :: Nil,
       newDefs = true))
   
-  final def exprOrIf(prec: Int, allowSpace: Bool = true)(implicit et: ExpectThen, fe: FoundErr, l: Line): IfBody \/ Term = wrap(prec, allowSpace) { l =>
+  private def letBindings(genQuote: Bool, l0: Loc)(implicit et: ExpectThen, fe: FoundErr, l: Line, qenv: Set[Str], quoted: IsQuoted): IfBody \/ Term = {
+    val bs = bindings(Nil)
+    val newQenv = if (genQuote) bs.map(_._1.name) else Nil
+    val body = yeetSpaces match {
+      case (KEYWORD("in" | ";;"), _) :: _ =>
+        consume
+        exprOrIf(0)(et, fe, implicitly, qenv ++ newQenv, quoted)
+      case (NEWLINE, _) :: _ =>
+        consume
+        exprOrIf(0)(et, fe, implicitly, qenv ++ newQenv, quoted)
+      case _ =>
+        R(UnitLit(true).withLoc(curLoc.map(_.left)))
+    }
+    bs.foldRight(body) {
```

Comment:
What's going on here? Why does parsing do this crazy stuff? Just always parse this as `Quoted(Let(false, v, Unquoted(r), Unquoted(acc)))`.